### DISCUSSION TITLE
Add ability to lock the modal state

### DISCUSSION THREAD

**Discussion by simpson** [tailwindlabs/headlessui#1257] [Ideas]
In some cases it would be preferable to lock the state of the modal to reduce the risk of a user losing data (Ex. form in modal). Can you add a prop that allows you to disable close when clicking anywhere in the overlay, perhaps even disabled the keyboard (esc) shortcut?

**Comment by Harrald** [comment]
An onBeforeClose event would be useful in which you could do your own checks and then stop the closing.

**Comment by mohammadzolfali5** [comment]
[That's perfect.]()

**Comment by simpson** [comment]
Why not just add a prop for this?
